packages/babel-preset-mapbox/index.js

```diff
@@ -122,8 +136,14 @@ module.exports = function preset(context, opts) {
 };
 
 function findProperty(data, name) {
-  const found = data.find(r => typeof r[0] === 'string' && r[0] === name);
-  return found && found[1];
+  if (data[name]) {
+    return data[name];
+  }
+  // NOTE: babel-preset & babel-plugin have same length
+  const len = 'babel-preset-'.length;
+  // If the user used a shorthand for eg. dynamic-import-node instead of babel-plugin-dynamic-import-node
+  // remove `babel-*-` and then try to find it.
+  return data[name.substring(len)];
```

Comment:
It might be more transparent to use a regex like `.replace(/^babel-(preset|plugin)-/, '')`?

Reply:
Ah, thats a great idea! I was wondering if there is a more readable approach to it.